tests/test_extras.c

```diff
@@ -33,14 +33,17 @@ int64_t cpucycles(void)
     return __rdpmccntr64();
 #elif (OS_TARGET == OS_LINUX) && (TARGET == TARGET_AMD64 || TARGET == TARGET_x86)
     unsigned int hi, lo;
-
-    asm volatile ("rdtsc\n\t" : "=a" (lo), "=d"(hi));
+    __asm__ volatile ("rdtsc\n\t" : "=a" (lo), "=d"(hi));
     return ((int64_t)lo) | (((int64_t)hi) << 32);
-#elif (OS_TARGET == OS_LINUX) && (TARGET == TARGET_ARM || TARGET == TARGET_ARM64)
+#elif (OS_TARGET == OS_LINUX) && (TARGET == TARGET_ARM || TARGET == TARGET_ARM64 )
     struct timespec time;
 
     clock_gettime(CLOCK_REALTIME, &time);
     return (int64_t)(time.tv_sec*1e9 + time.tv_nsec);
+#elif (OS_TARGET == OS_LINUX && TARGET == TARGET_S390X)
+    uint64_t tod;
+    __asm__ volatile("stck %0\n" : "=Q" (tod) : : "cc");

```

Comment:
```suggestion
    __asm__ volatile("stckf %0\n" : "=Q" (tod) : : "cc");

```

stckf would be better here than stck.  stck serializes the clock across all CPUs to guarantee system unique timestamps.

Reply:
Thank you for the suggestion. Added a commit and changed it to stckf.
From what I read, stckf is available from z9. Since we are compiling for z10 it should be safe to use.